### PR TITLE
Replace User hbm mapping file with annotations

### PR DESCRIPTION
**Pull request by sherrif10**
Replaced user hbm mapping with annotation
  The  ticket link 
https://issues.openmrs.org/browse/TRUNK-5664

### CODE REVIEW

**Comment by sherrif10** [comment]
@ Dimitri R, @mozzy11  kindly review my PR thanks

**Comment by sherrif10** [comment]
@mks-d kindly review this PR thanks

**Comment by mozzy11** [comment]
@sherrif10  Thanks for the work done , it makes a lot of sense
However take note of the following
* Squah your commits into one ?
* and also have you seen the merge comflicts ?
* You should also create a PR from a branch not your master. 
see https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/src/main/java/org/openmrs/User.java [hunk 1]
```diff
@@ -30,49 +43,111 @@
...
+		this.userId = userId;
+		this.person = person;
+		this.systemId = systemId;
+		this.username = username;
+		this.email = email;
+		this.roles = roles;
+		this.userProperties = userProperties;
+		this.proficientLocales = proficientLocales;
+		this.parsedProficientLocalesProperty = parsedProficientLocalesProperty;
+	}
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/User.java]: What's this new constructor about?

**Comment by mozzy11** [comment]
@sherrif10  , Unfortunately @wluyima  is already working on this ticket. Please ensure you find tickets unclaimed to work on , and always ensure to claim them.  Thanks

**Comment by mozzy11** [comment]
@sherrif10  you can please close this PR

**Comment by wluyima** [comment]
Closing this as this has already been worked on, see https://github.com/openmrs/openmrs-core/pull/3027

**Comment by sherrif10** [comment]
Thanks,  have already  closed it

**Pull request closed without merging**